src/TierRegistry.sol

```diff
@@ -0,0 +1,330 @@
+import {Pausable} from "openzeppelin/contracts/security/Pausable.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import "openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {ITierRegistry} from "./interfaces/ITierRegistry.sol";
+import {TransferHelper} from "./libraries/TransferHelper.sol";
+
+/**
+ * @title Farcaster StorageRegistry
+ *
+ * @notice See https://github.com/farcasterxyz/contracts/blob/v3.1.0/docs/docs.md for an overview.
+ *
+ * @custom:security-contact security@merklemanufactory.com
+ */
+contract TierRegistry is ITierRegistry, AccessControlEnumerable, Pausable {
+    using FixedPointMathLib for uint256;
+    using TransferHelper for address;
+    using SafeERC20 for IERC20;
+
+    /*//////////////////////////////////////////////////////////////
+                                 ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Revert if the caller attempts to rent storage after the contract is deprecated.
+    error ContractDeprecated();
+
+    /// @dev Revert if the caller attempts to rent zero units.
+    error InvalidAmount();
+
+    error InvalidTier();
+
+    error InvalidPrice();
+
+    error InvalidBatchInput();
+
+    error InvalidAddress();
+
+    /// @dev Revert if the caller is not an owner.
+    error NotOwner();
+
+    /// @dev Revert if the caller is not an operator.
+    error NotOperator();
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    // TODO(aditi): Maybe rename without Purchase in name
+    event PurchasedTier(uint256 indexed fid, uint256 tier, uint256 forDays);
+
+    /**
+     * @dev Emit an event when an owner changes the vault.
+     *
+     * @param oldVault The previous vault.
+     * @param newVault The new vault.
+     */
+    event SetVault(address oldVault, address newVault);
+
+    event SetToken(address oldToken, address newToken);
+
+    event SetTierPrice(uint256 tier, address token, uint256 oldPrice, uint256 newPrice);
+
+    event SetMinDays(uint256 oldMinDays, uint256 newMinDays);
+
+    event SetMaxDays(uint256 oldMaxDays, uint256 newMaxDays);
+
+    /*//////////////////////////////////////////////////////////////
+                                CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc ITierRegistry
+     */
+    string public constant VERSION = "2025.05.21";
+    // TODO(aditi): Update the date
+
+    bytes32 internal constant OWNER_ROLE = keccak256("OWNER_ROLE");
+    bytes32 internal constant OPERATOR_ROLE = keccak256("OPERATOR_ROLE");
+
+    /*//////////////////////////////////////////////////////////////
+                              PARAMETERS
+    //////////////////////////////////////////////////////////////*/
+
+    uint256 public minDays;
```

Comment:
In order to make monthly the min, we can set minDays to 30. Days felt like a better unit to denominate by in the contract because month lengths are nonstandard.